src/lib.rs

```diff
@@ -0,0 +1,555 @@
+///     7. edges: `vec` for storing edges which reading from the edge file with separate,
+///     8. neighbour_index: `HashMap` for indexing node's neighbours pos in `vec`.
+///
+/// Here are the descriptions for the functions:
+///     1. default(): create a default graph data;
+///     2. with_dir(GraphType::Directed(or Undirected)): the graph constructing is directed or not;
+///     3. with_schema(): identifing the schema of the graph;
+///     4. from_file(edge_file:String,node_file:Option<String>,sep:String): node_file path,
+///             and optional edge file path, both are csv files where each term is separated with sep.
+///     5. build(): return a graph instance according to the iniformations gave above.
+///
+/// # Example
+///
+/// ```
+///  use graph_rust::{Graph, GraphType};
+///  let graph = Graph::default()
+///        .with_dir(GraphType::Directed)
+///        .with_schema(String::from("..."))
+///        .from_file(String::from("data_edges.csv"), None, ",")
+///        .build();
+/// ```
+#[derive(Clone)]
+pub struct Graph<T> {
+    graph_type: GraphType,
+    schema: String,
+    node_file: Option<String>,
+    edge_file: String,
+    separate: String,
+    vertices: Vec<Vertex<T>>,
+    edges: Vec<Edge<T>>,
+    neighbour_index: HashMap<T, (usize, usize)>,
+}
+
+impl<T> Graph<T>
+    where
+        T: Eq + Hash + FromStr + Display + Clone + Copy,
+        <T as FromStr>::Err: Debug,
+{
+    pub fn default() -> Graph<T> {
+        return Graph {
+            graph_type: GraphType::Directed,
+            schema: "".to_string(),
+            node_file: None,
+            edge_file: "".to_string(),
+            separate: "".to_string(),
+            vertices: vec![],
+            edges: vec![],
+            neighbour_index: HashMap::new(),
+        };
+    }
+
+    pub fn with_dir(&mut self, dir: GraphType) -> &mut Graph<T> {
+        self.graph_type = dir;
+        self
+    }
+
+    pub fn with_schema(&mut self, schema: String) -> &mut Graph<T> {
+        self.schema = schema;
+        self
+    }
+
+    pub fn from_file(
+        &mut self,
+        edge_file: String,
+        node_file: Option<String>,
+        sep: &str,
+    ) -> &mut Graph<T> {
+        self.node_file = node_file;
+        self.edge_file = edge_file;
+        self.separate = sep.to_string();
+        self
+    }
+
+    pub fn build(&mut self) -> Graph<T> {
+        //Init node and edge container
+        let mut vertices = HashMap::new();
+        let mut edges = HashMap::new();
+
+        let directed = match self.graph_type {
+            GraphType::Directed => true,
+            GraphType::Undirected => false,
+        };
+
+        let no_node_file = match self.node_file {
+            None => true,
+            _ => false,
+        };
+
+        // Open file and read line by line
+        let edge_file = File::open(&self.edge_file).expect("Please check `edge_file` path");
```

Comment:
I think you can use a library called `CsvReader` to read the files. Note that the csv may maintain a header as the attributes' name so that you do not need to specify something like `a=1,b=1,c=1`, but rather 
```
id,a,b,c  // This is the header
0,1,1,1   
1,2,2,2
...
```
Use that header to parse schema